### PR TITLE
Disable heartbeat if duration set to zero

### PR DESCRIPTION
**Pull request by jwmdev**
## Description

Disable sync hearbeat if heartbeat dealy is set to zero. This also prevents hearbeat ticker from panic if the duration is zero.

## Related issue(s)

Disable heartbeat #164 

## Checklist

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] CHANGELOG is updated.

### CODE REVIEW

File: sync/config.go [hunk 1]
```diff
@@ -8,6 +8,9 @@ import (
 
 var LatestBlockInterval = uint32(720) // 720 blocks is about two hours
 
+// IMPORTANT NOTE:
```
Comment (b00f) [inline sync/config.go:11]: This comment here is not useful. Code is very clear. 
We should inform end user. Please remove this comment and update example: https://github.com/zarbchain/zarb-go/blob/main/node/config/example_config.toml

Example config will be created if user create a new node.

File: sync/config.go [hunk 2]
```diff
@@ -23,7 +26,7 @@ type Config struct {
 func DefaultConfig() *Config {
 	return &Config{
 		StartingTimeout:  time.Second * 3,
-		HeartBeatTimeout: time.Second * 5,
+		HeartBeatTimeout: time.Second * 5, //set this value to zero if you want to disable heart beat
```
Comment (b00f) [inline sync/config.go:29]: This comment here also not necessary. Code is clear.

**Review by b00f — CHANGES_REQUESTED** [review]

**Comment by b00f** [comment]
@jwmdev I noticed that your branch is from old p2p branch. Look at the commits here. It's recommended to create new branch from the latest main branch.

**Comment by jwmdev** [comment]
improved the comments... I will update the repo after merging this PR

File: sync/sync.go [hunk 3]
```diff
@@ -121,7 +123,9 @@ func (sync *synchronizer) Start() error {
 
 func (sync *synchronizer) Stop() {
 	sync.ctx.Done()
-	sync.heartBeatTicker.Stop()
+	if sync.config.HeartBeatTimeout > 0 { //this is to prevent calling Stop on a null heartBeatTicker
```
Comment (b00f) [inline sync/sync.go:126]: I suggest change this condition to: `if sync.heartBeatTicker != nil`

**Review by b00f — CHANGES_REQUESTED** [review]
@jwmdev Better to check if heartBeatTicker is not nil to stop it.

**Comment by b00f** [comment]
@jwmdev LGTM. Thank you so much.

**Review by b00f — APPROVED** [review]

**Pull request merged**